docs/database/includes/azure-postgresql-ef-client.md

```diff
@@ -21,31 +21,48 @@ dotnet add package Azure.Identity
 
 ---
 
-The PostgreSQL connection can be consumed using the client integration and <xref:Azure.Identity>:
+The PostgreSQL connection can be consumed using the client integration and <xref:Azure.Identity>.
+
+The following code snippets demonstrate how to use the <xref:Azure.Identity.DefaultAzureCredential> class from the <xref:Azure.Identity> package to authenticate with [Microsoft Entra ID](/azure/postgresql/flexible-server/concepts-azure-ad-authentication) and retrieve a token to connect to the PostgreSQL database. The [UsePasswordProvider](https://www.npgsql.org/doc/api/Npgsql.NpgsqlDataSourceBuilder.html#Npgsql_NpgsqlDataSourceBuilder_UsePasswordProvider_System_Func_Npgsql_NpgsqlConnectionStringBuilder_System_String__System_Func_Npgsql_NpgsqlConnectionStringBuilder_System_Threading_CancellationToken_System_Threading_Tasks_ValueTask_System_String___) method is used to provide the token to the data source builder.
+
+### EntityFramework Core version 8
 
 ```csharp
-builder.AddNpgsqlDbContext<YourDbContext>(
-    "postgresdb", 
-    configureDataSourceBuilder: (dataSourceBuilder) =>
+var dsBuilder = new NpgsqlDataSourceBuilder(builder.Configuration.GetConnectionString("postgresdb"));
+if (string.IsNullOrEmpty(dsBuilder.ConnectionStringBuilder.Password))
 {
-    if (!string.IsNullOrEmpty(dataSourceBuilder.ConnectionStringBuilder.Password))
-    {
-        return;
-    }
-
-    dataSourceBuilder.UsePeriodicPasswordProvider(async (_, ct) =>
-    {
-        var credentials = new DefaultAzureCredential();
-        var token = await credentials.GetTokenAsync(
-            new TokenRequestContext([
-                "https://ossrdbms-aad.database.windows.net/.default"
-            ]), ct);
-
-        return token.Token;
-    },
-    TimeSpan.FromHours(24),
-    TimeSpan.FromSeconds(10));
-});
+    var credentials = new DefaultAzureCredential();
+    var tokenRequest = new TokenRequestContext(["https://ossrdbms-aad.database.windows.net/.default"]);
+
+    dsBuilder.UsePasswordProvider(
+        passwordProvider: _ => credentials.GetToken(tokenRequest).Token,
+        passwordProviderAsync: async (_, ct) => (await credentials.GetTokenAsync(tokenRequest, ct)).Token);
+}
+
+builder.AddNpgsqlDbContext<MyDb1Context>(
+    "postgresdb",
+    configureDbContextOptions: (options) => options.UseNpgsql(dsBuilder.Build()));
+```
+
+### EntityFramework Core version 9+
+
+With EntityFramework Core version 9, you can use the new `ConfigureDataSource` method to configure the NpgsqlDataSourceBuilder that is used by the integration instead of building one outside of the integration and passing it in.
```

Comment:
```suggestion
With EF Core version 9, you can use the `ConfigureDataSource` method to configure the `NpgsqlDataSourceBuilder` that's used by the integration instead of building one outside of the integration and passing it in.
```